### PR TITLE
Add custom grid example

### PR DESCRIPTION
**Pull request by ds300**
#4420 highlighted for me that we don't have an example to demonstrate how to implement a custom grid, and this is definitely a common enough need to warrant an example IMO.

### Change type

- [x] `improvement`

### CODE REVIEW

File: apps/examples/src/examples/custom-grid/CustomGridExample.tsx
Comment (mimecuvalo) [inline apps/examples/src/examples/custom-grid/CustomGridExample.tsx:21]: is `useLayoutEffect` necessary here? it seems like `useEffect` would suffice but lemme know if i'm missing something

**Review by mimecuvalo — APPROVED** [review]
LG! pre-approving modulo one comment about `useLayoutEffect` before landing though

File: apps/examples/src/examples/custom-grid/CustomGridExample.tsx
Comment (ds300) [reply apps/examples/src/examples/custom-grid/CustomGridExample.tsx:21]: i think it is yeah. With `useEffect` the grid would lag behind the other canvas elements by one frame?

Comment (mimecuvalo) [reply apps/examples/src/examples/custom-grid/CustomGridExample.tsx:21]: I think that might be ok btw to be one frame behind. Esp. for slower machines, it makes me wonder if we should be encouraging this pattern of strong consistency vs. something that might make the rendering slower potentially.

Comment (ds300) [reply apps/examples/src/examples/custom-grid/CustomGridExample.tsx:21]: nah i think it's fine. drawing a few thousand lines on a canvas is so fast it's negligible.

**Pull request merged**
